src/shared/mount-util.c

```diff
@@ -474,6 +474,44 @@ int bind_remount_one_with_mountinfo(
         return 0;
 }
 
+int mount_pivot_root(const char *path) {
+        _cleanup_close_ int fd_oldroot = -1, fd_newroot = -1;
```

Comment:
so far we always assign -1 to invalid fds. But maybe we should follow your example there and use -EBADF instead, it does make sense, so please leave this as it is.

but fd_newroot needs the same treatment

Reply:
ah, you already fixed this in a newer push. ignore this then.